### PR TITLE
Change string replace operators to use StringUtils from apache library

### LINKED ISSUE #57: Problem using ReplaceFirstSubSeq and ReplaceAllSubSeqs with strings
Hello, the replace operator with strings is not correctly replacing the following chars: "\n", "\t", "\f", and"\r".

For example, in the following assume expressions, the first two are true and the last four false.

```tla
ASSUME(ReplaceAllSubSeqs("\\", "%%", "Properly escape the %% quotes") = "Properly escape the \\ quotes")
ASSUME(ReplaceAllSubSeqs("\"", "%%", "Properly escape the %% quotes") = "Properly escape the \" quotes")
ASSUME(ReplaceAllSubSeqs("\n", "%%", "Properly escape the %% quotes") = "Properly escape the \n quotes")
ASSUME(ReplaceAllSubSeqs("\t", "%%", "Properly escape the %% quotes") = "Properly escape the \t quotes")
ASSUME(ReplaceAllSubSeqs("\f", "%%", "Properly escape the %% quotes") = "Properly escape the \f quotes")
ASSUME(ReplaceAllSubSeqs("\r", "%%", "Properly escape the %% quotes") = "Properly escape the \r quotes")
```

Some options that i found that can fix it are:

1. Not using unquote and not using Pattern.quote.

Example: 
((StringValue) t).toUnquotedString(); -> ((StringValue) t).getVal().toString();
st.replaceFirst(Pattern.quote(ss), sr) -> st.replaceAll(ss, sr)
Problem:
Slash and dollar sign have to be special escaped, for example the first assume expression would have to change to:
` ASSUME(ReplaceAllSubSeqs("\\\\", "%%", "Properly escape the %% quotes") = "Properly escape the \\ quotes")`
However, the remain ones would work as they are.

2. Change to use functions without regex.

For replaceAll there is replace. For replaceFirst I did not find one in the standard library, however there is replaceOnce in org.apache.commons.lang3.StringUtils that can be used. There is also the option to create a custom one using find, substr and replace.

I can work on a pull request for this issue, what would be the best approach to modify ReplaceFirstSubSeq and ReplaceAllSubSeqs ?

## REF
https://github.com/tlaplus/CommunityModules/blob/master/modules/tlc2/overrides/SequencesExt.java#L243
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replace(java.lang.CharSequence,%20java.lang.CharSequence)
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replaceAll(java.lang.String,%20java.lang.String)
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replaceFirst(java.lang.String,%20java.lang.String)
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#replaceOnce-java.lang.String-java.lang.String-java.lang.String-

### PR DESCRIPTION
**Pull request by afonsonf**
Change how replace is done to use StringUtils from apache library.
Add tests for all special characters of TLA+ in strings.
Updated some tests because the new library has different behaviour in edge cases: e.g. replace "" with "a" in "" equals "".

Update:
Now replace "" with "a" in "" equals "a".

Closes #57

### CODE REVIEW

**Comment by lemmy** [comment]
Does the "edge case" violate the TLA+ def of the operator?

**Comment by afonsonf** [comment]
> Does the "edge case" violate the TLA+ def of the operator?

Yes, however I changed it now to match the TLA+ operator behaviour and reverted the tests

**Comment by lemmy** [comment]
LGTM (generally I'd suggest adding curly braces even to single-line if statements).

File: modules/tlc2/overrides/SequencesExt.java [hunk 1]
```diff
@@ -266,11 +268,20 @@ public static Value replaceAllSubSeq(final Tool tool, final ExprOrOpArgNode[] ar
...
-			final String ss = ((StringValue) s).toUnquotedString();
-			final String sr = ((StringValue) r).toUnquotedString();
+			final String st = ((StringValue) t).getVal().toString();
+			final String ss = ((StringValue) s).getVal().toString();
+			final String sr = ((StringValue) r).getVal().toString();
+			
+			if(ss.equals("")) {
+				StringBuilder result = new StringBuilder(sr);
+				for(int i=0;i<st.length();i++) {
+					if(i != 0) result.append(sr);
```
Comment (lemmy) [inline modules/tlc2/overrides/SequencesExt.java:278]: Add `{}`?

**Review by lemmy — APPROVED** [review]

File: modules/tlc2/overrides/SequencesExt.java (hunk 1 above)
Comment (afonsonf) [reply modules/tlc2/overrides/SequencesExt.java]: Missed that one 😅 Done, thanks for the review!

**Comment by lemmy** [comment]
Are you going to squash and merge to master?
